pkg/tasks/kubestatemetrics.go

```diff
@@ -85,11 +93,55 @@ func (t *KubeStateMetricsTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "reconciling kube-state-metrics Service failed")
 	}
 
+	cm, err := t.factory.KubeStateMetricsCRSConfigMap()
+	if err != nil {
+		return errors.Wrap(err, "initializing kube-state-metrics custom-resource-state ConfigMap failed")
+	}
+
+	err = t.client.CreateOrUpdateConfigMap(ctx, cm)
+	if err != nil {
+		return errors.Wrapf(err, "reconciling %s/%s ConfigMap failed", cm.Namespace, cm.Name)
+	}
+
 	dep, err := t.factory.KubeStateMetricsDeployment()
 	if err != nil {
 		return errors.Wrap(err, "initializing kube-state-metrics Deployment failed")
 	}
 
+	// Check if custom-resource-state metrics have been enabled.
+	// * If so, add the --custom-resource-state-config-file flag to the kube-state-metrics container, if it is absent.
+	// * If not, remove the --custom-resource-state-config-file flag from the kube-state-metrics container, if it is present.
+	// This will, in turn, also cause the kube-state-metrics container to be restarted with the new set of flags in effect.
+	if t.enableCRSMetrics {
```

Comment:
it should be moved to t.factory.KubeStateMetricsDeployment() as we do for other components. You can add tests to pkg/manifests/manifests_test.go